IOPool/Input/src/RepeatingCachedRootSource.cc

```diff
@@ -228,13 +228,17 @@ RepeatingCachedRootSource::RepeatingCachedRootSource(ParameterSet const& pset, I
   }
 }
 
-void RepeatingCachedRootSource::beginJob() {
+void RepeatingCachedRootSource::beginJob(ProductRegistry const&) {
   ProcessConfiguration processConfiguration;
   processConfiguration.setParameterSetID(ParameterSet::emptyParameterSetID());
   processConfiguration.setProcessConfigurationID();
 
+  //in order to use the source's internal ProductRegistry for looking up date
+  // it needs to be frozen (which setups the other structures)
+  productRegistryUpdate().setFrozen();
   //Thinned collection associations are not supported at this time
-  EventPrincipal eventPrincipal(productRegistry(),
+  auto noDel = [](void const*) {};
+  EventPrincipal eventPrincipal(std::shared_ptr<ProductRegistry const>(&productRegistry(), noDel),
```

Comment:
How about using
https://github.com/cms-sw/cmssw/blob/a621796a69a26ce6804093f920578b6e1dce3dfd/FWCore/Utilities/interface/do_nothing_deleter.h#L33-L35
instead of `noDel`?

Reply:
done